libraries/AP_HAL_Linux/Util_RPI.cpp

```diff
@@ -27,33 +28,53 @@ extern const AP_HAL::HAL &hal;
 
 using namespace Linux;
 
-UtilRPI::UtilRPI()
-{
-    _get_board_type_using_peripheral_base ();
-}
-
+UtilRPI::UtilRPI() { _get_board_type_using_peripheral_base(); }
 
-// 
+//
 // previous appraoch was using /proc/device-tree/system/linux,revision
-// now we use /proc/device-tree/soc/ranges see: https://forums.raspberrypi.com//viewtopic.php?t=244031
-void UtilRPI::_get_board_type_using_peripheral_base() 
-{
+// now we use /proc/device-tree/soc/ranges see:
+// https://forums.raspberrypi.com//viewtopic.php?t=244031
+void UtilRPI::_get_board_type_using_peripheral_base() {
     FILE *fp;
-    uint32_t base=0x00;
+    uint32_t base = 0x00;
     unsigned char buf[32];
     _linux_board_version = LINUX_BOARD_TYPE::UNKNOWN_BOARD;
-    fp = fopen("/proc/device-tree/soc/ranges" , "rb");
+
+    const char *base_path = "/proc/device-tree";
+    DIR *dir = opendir(base_path);
+    if (!dir) {
+        printf("device-tree directory not found \r\n");
+        return;
+    };
+
+    struct dirent *entry;
+    char ranges_path[256] = {0};
+
+    while ((entry = readdir(dir)) != nullptr) {
+        if (strncmp(entry->d_name, "soc", 4) == 0) {
+            snprintf(ranges_path, sizeof(ranges_path), "%s/%s/ranges",
+                     base_path, entry->d_name);
+            break;
+        }
+    }
+    closedir(dir);
+
+    if (ranges_path[0] == 0) {
+        printf("\"ranges\" file not found \r\n");
+        return;
+    };
+
+    fp = fopen(ranges_path, "rb");
     if (fp) {
         const uint16_t len = fread(buf, 1, sizeof(buf), fp);
         if (len >= 8) {
-            base = buf[4]<<24 | buf[5]<<16 | buf[6]<<8 | buf[7];
+            base = buf[4] << 24 | buf[5] << 16 | buf[6] << 8 | buf[7];
```

Comment:
Not a fan of this whitespace change, grouping the shift bits together is better.